src/convert.rs

```diff
@@ -27,3 +30,24 @@ impl TryFrom<usize> for StrIndex {
         i.try_into().map(|raw| StrIndex { raw })
     }
 }
+
+impl<B> From<B> for StrRange
+where
+    B: RangeBounds<StrIndex>,
+{
+    fn from(bounds: B) -> Self {
+        let start = match bounds.start_bound() {
+            Bound::Included(i) => *i,
+            Bound::Excluded(i) => i + StrIndex::from(1),
```

Comment:
`StrIndex::from(1)` is wrong. It should be `StrIndex::from_char(char_at_position_i)`, which we can't express without str

Reply:
Given that this case isn't supported by `std` indexing (that I can find with a quick overview),  I don't actually know what the behavior should be in this case. The end case would suggest that it's just "exclude the byte at this index", but I don't know.

I've half a mind to only support `Range` and `RangeTo`, and ignore the other `RangeBounds` types for simplicity.